File: solutions/go/392.go
func isSubsequence(s string, t string) bool {
	if len(s) > len(t) {
		return false
	}

	var idx int

	for i := 0; i < len(t); i++ {
		if idx < len(s) && s[idx] == t[i] {
			idx++
		}
	}

	return idx == len(s)
}

File: solutions/rust/392.rs
impl Solution {
    pub fn is_subsequence(s: String, t: String) -> bool {
        if s.len() > t.len() {
            return false;
        }

        let s = s.as_bytes();
        let t = t.as_bytes();
        let mut idx = 0;

        for b in t {
            if idx < s.len() && s[idx] == *b {
                idx += 1;
            }
        }

        idx == s.len()
    }
}
